packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -147,14 +142,26 @@ export const useCheckoutPage = props => {
     ]);
 
     useEffect(() => {
-        if (cartId) {
+        const placedOrder = !!placeOrderData;
```

Comment:
I was expecting to see the `called` property of the mutation used here, not the presence of `data`.

This opens up a window between `called` and `data` where the `cartId` could change and this bug could resurface (say, if the place order network call takes a while).

Reply:
Nice point @supernova-at. But my doubt is, what happens if the `place order mutation` is called but fails? In that case `called` will be true but that is a false positive for our code. We only want to skip doing this if the mutation is successful.

Also, I don't think the cart id will be cleared without the place order mutation being a success, because of the way the `handlePlaceOrder` is written. `createCart` will await till `placeOrder` is successful.

```es6
    () => {
        await getOrderDetails({
            variables: {
                cartId
            }
        });

        await placeOrder({
            variables: {
                cartId
            }
        });

        await removeCart();

        await clearCartDataFromCache(client);

        await createCart({
            fetchCartId
        });
    }
```

Reply:
Yeah, I was thinking about that too ... 

If you leave it as is you may want to leave a comment, mostly because the block comment below it says "wait until place order is **called**".

Reply:
Not relevant anymore because the loading logic has changed.